### DISCUSSION TITLE
manage multiple pictures

### DISCUSSION THREAD

**Discussion by misterpatate** [pgrimaud/instagram-user-feed#227] [Q&A]
I may have missed it in the API but I can't find a way to manage posts with several pictures.

Thank you for your API.

<img width="749" alt="uniqlo" src="https://user-images.githubusercontent.com/8451544/128850629-01f8bc6c-fadb-430b-8144-460a10346006.PNG">

**Comment by pgrimaud** [comment]
Unfortunately, you can't fetch all pictures with several pictures when using the media feed.

You have to fetch the post to retrieve all pictures of a post.

**Comment by OzzWizard** [comment]
I found such a solution for myself. Maybe something better can be done.
```php
function albumGetir($url,$api)
{
    $a = [];
    $media = new Media();
    $media->setLink("https://instagram.com/p/".$url);
    $mediaDetailed = $api->getMediaDetailed($media);
    $count = count($mediaDetailed->getSideCarItems());
    for($i = 1; $i<=$count;$i++)
    {
	$test = $mediaDetailed->getSideCarItems()[$i-1]->getVideoUrl();
        if(!$test)
         $a += [$i => ["url" => end($mediaDetailed->getSideCarItems()[$i-1]->getDisplayResources())->src]];
        else
         $a += [$i => ["url" =>$mediaDetailed->getSideCarItems()[$i-1]->getVideoUrl()]];
    }
	return json_encode($a);
}
